uniffi_bindgen/src/bindings/swift/templates/ErrorTemplate.swift

```diff
@@ -83,12 +83,16 @@ public enum {{e.name()}}: RustError {
         switch rustError.code {
             case 0:
                 return nil
+            case -1:
+            // TODO: Return a Uniffi defined error here
+            // to indicate a panic
+                fatalError("Panic detected!")
```

Comment:
Aye, this is a broader thing we have to think about in the Swift bindings - should it be possible to catch and handle Rust panics? And does it make sense that you can do this only for functions that otherwise return an error? I'm not familiar enough with consuming the swift bindings to know what a reasonable set of expectations are.

Reply:
Hmm it seems functions that don't return errors capture the errors, but instead of returning them to the consumer, [they cause a fatal error because of the `try!` here](https://github.com/mozilla/uniffi-rs/blob/main/uniffi_bindgen/src/bindings/swift/templates/macros.swift#L100) So that adds a bit of context, but still isn't exposed to the consumer... So maybe the same approach? But ya, other than messing around with it in this project, my Swift knowledge is nil 🙃

Reply:
Right, and this is kind of unfortunately necessary because of Swift's explicit checked-exceptions kind of model. We either force *every* call to a Swift API function to have to explicitly handle the possibility of an error, or we allow functions that "should" never fail to be called without a `try` handler and are forced to fatally error out if they do fail in practice.

(It's similar in a lot of ways to Rust's `Result` type; if we were generating Rust bindings, would we want to force every single exposed function and method to return a `Result` just to be able to handle the case of a panic?)

Reply:
Yeah I'm a fan of the "let them fatally error" approach, seems the most "true" to Rust's approach. Because those types of errors should only happen if we observed a panic (which would have fatally errored anyways) or something went awfully wrong. I'll add this bit of context to #487